bigtable/google/cloud/bigtable/row.py

```diff
@@ -409,28 +407,13 @@ def commit(self):
         :raises: :class:`ValueError <exceptions.ValueError>` if the number of
                  mutations exceeds the :data:`MAX_MUTATIONS`.
         """
-        mutations_list = self._get_mutations(None)
-        num_mutations = len(mutations_list)
+        num_mutations = len(self.row_mutations.mutations)
         if num_mutations == 0:
-            return
+            return {}
         if num_mutations > MAX_MUTATIONS:
-            raise ValueError('%d total mutations exceed the maximum allowable '
-                             '%d.' % (num_mutations, MAX_MUTATIONS))
-
-        data_client = self._table._instance._client.table_data_client
-        commit = functools.partial(
-            data_client.mutate_row,
-            self._table.name, self._row_key, mutations_list)
-        retry_ = retry.Retry(
-            predicate=_retry_commit_exception,
-            deadline=30)
-        retry_(commit)()
-
-        self.clear()
-
-    def clear(self):
-        """Removes all currently accumulated mutations on the current row."""
-        del self._pb_mutations[:]
+            raise ValueError('%d total append mutations exceed the maximum '
```

Comment:
This error should be thrown from `row_mutations` methods instead of doing the logic here.